nav2_costmap_2d/test/unit/collision_footprint_test.cpp

```diff
@@ -0,0 +1,94 @@
+#include <string>
+#include <vector>
+#include <memory>
+
+#include "gtest/gtest.h"
+#include "nav2_costmap_2d/collision_footprint.hpp"
+
+TEST(collision_footprint, footprintCostWithPose)
```

Comment:
These tests should more accurately test that things are working, having an empty costmap is going to always work. Can you add some obstacles around the robot but just barely not touching to make it try to strain things if anything is really wrong?

Reply:
A test with obstacles all around it but not touching it, move the footprint over 1 pixel, and check that its now in collision.